components/patient-measurements/api/src/main/java/org/phenotips/measurements/internal/MeasurementAgeUpdater.java

```diff
@@ -95,6 +93,8 @@ public void onEvent(Event event, Object source, Object data)
             Date measurementDate = measurement.getDateValue(DATE_PROPERTY_NAME);
             if (measurementDate == null) {
                 measurement.removeField(AGE_PROPERTY_NAME);
+            } else if (birthDate == null) {
```

Comment:
Since the same action is performed as in the above branch, update the condition:

    if (measurementDate == null || birthDate == null) {